project/src/main/java/com/google/autograder/servlets/ListAssignmentsServlet.java

```diff
@@ -64,4 +70,86 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         }
     }
 
+    private void storeAssignmentsDataInDatastore(String courseJSON, String courseID) {
+        try {
+            Filter courseIDFilter = new FilterPredicate("courseID", FilterOperator.EQUAL, courseID);
+            Query query = new Query("Assignment").setFilter(courseIDFilter);
+            PreparedQuery results = Services.DATA_STORE.prepare(query);
+
+            for (Entity assignment : results.asIterable()) {
+                Services.DATA_STORE.delete(assignment.getKey());
+            }
+
+            JSONObject courseWorkObject = (JSONObject) new JSONParser().parse(courseJSON);; 
+            JSONArray courseWork = (JSONArray) courseWorkObject.get("courseWork");
+            Iterator courseWorkIterator = courseWork.iterator();
+
+            while (courseWorkIterator.hasNext()) {
+
+                JSONObject courseWorkTempObject = (JSONObject) courseWorkIterator.next();
+                String courseWorkType = (String) courseWorkTempObject.get("workType");
+
+                if (courseWorkType.equals("ASSIGNMENT")) {
+                    JSONObject assignment = courseWorkTempObject;
+
+                    String id = (String) assignment.get("id");
+                    String title = (String) assignment.get("title");
+                    long maxPoints = (long) assignment.get("maxPoints");
+                    String description = (String) assignment.get("description");
+                    String creationTime = (String) assignment.get("creationTime");
+
+                    Entity assignmentEntity = new Entity("Assignment");
+
+                    assignmentEntity.setProperty("id", id);
```

Comment:
Didn't Mahima make a function to do this?

Reply:
Yeah, but it was slightly different than what I needed here. We talked about it and I made this function here and then we're gonna consolidate it into the Database class in the next PR.

Reply:
+1, would be great to have the DATA_STORE return a POJO rather than a plain entity.

Reply:
Yeah the entity class is a little weird, it's more like a map than an actual object.